web/files/files.go

```diff
@@ -134,11 +192,58 @@ func ReadHandler(c *gin.Context) {
 
 // Routes sets the routing for the files service
 func Routes(router *gin.RouterGroup) {
-	router.HEAD("/:file-id", ReadHandler)
-	router.GET("/:file-id", ReadHandler)
+	router.HEAD("/:file-id", ReadFileHandler)
+	router.GET("/:file-id", ReadFileHandler)
 
 	router.POST("/", CreationHandler)
 	router.POST("/:folder-id", CreationHandler)
+
+	router.PUT("/:file-id", OverwriteFileContentHandler)
+}
+
+func getFsAndDBPrefix(c *gin.Context) (fs afero.Fs, dbPrefix string, err error) {
+	instance := middlewares.GetInstance(c)
+	dbPrefix = instance.GetDatabasePrefix()
+	fs, err = instance.GetStorageProvider()
```

Comment:
[Future] May be all this instanciating of request specific structs (instance, dbprefix, fs) could be moved to middlewares and c.MustGet.